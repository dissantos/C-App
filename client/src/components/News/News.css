.News {
    padding: 20px;
}

.subtitle {
    margin: 0;
    padding: 0;
    padding: 0 15rem;
    text-align: left;
    color: gray;
}

.date {

    padding-left: 15rem;
    text-align: left;
}

.content p {
    margin: 0;
    padding: 0;
    padding: 0 15rem;
    margin-block-end: 15px;
    width: auto;
    text-align: justify;
}

.content p img {
    display: block;
    padding: 5px;
}