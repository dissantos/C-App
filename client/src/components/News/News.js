import React, { useEffect, useState } from 'react';
import './News.css';
import Divider from '@mui/material/Divider';
import Typography from '@material-ui/core/Typography';
import parse from 'html-react-parser';
import getNoticia from "../../functions/getNoticia";

const News = (props) => {
  const params = window.location.href.split("/");
  const path = params[params.length - 2];
  const [state, setState] = useState({
    title: '',
    subtitle: '',
    date: '',
    content: []
  });


  useEffect(() => {
    const fetchNews = async () => {
      const response = await getNoticia(path);
      const newsContent = {
        title: response.title,
        subtitle: response.subtitle,
        date: response.date,
        content: response.content
      }
      setState(newsContent)
    }
    fetchNews();
  },[])

  return (
    <div className="News">
      <Divider textAlign="center">
        <Typography variant='h5' sx={{padding: '20px', color: '#5579b9'}}>{state.title}</Typography>
      </Divider>
      <h3 className="subtitle">{state.subtitle}</h3>
      <h5 className="date">{state.date}</h5>
      <div className="content">
        {state.content.map(el => {
          return parse(el)
        })}
      </div>
    </div>
  );
}

News.propTypes = {};

News.defaultProps = {};

export default News;
