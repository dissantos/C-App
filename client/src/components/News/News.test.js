import React from 'react';
import News from './News';
import toJson from 'enzyme-to-json'
import { mount } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() })

it('should render News page', () => {
    const spy = jest.spyOn(global.console, 'error')
    
    const app = mount( 
        <News category="Graduação" title="Edital seleciona alunos especiais para Mestrado em Automação e Sistemas" url="https://www.cefetmg.br/noticias/edital-seleciona-alunos-especiais-para-mestrado-em-automacao-e-sistemas/"/>,
    )
    expect(toJson(app)).toMatchSnapshot()
    expect(spy).not.toHaveBeenCalled()
})