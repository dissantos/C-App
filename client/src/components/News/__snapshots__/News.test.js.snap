// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render News page 1`] = `
<News
  category="Graduação"
  title="Edital seleciona alunos especiais para Mestrado em Automação e Sistemas"
  url="https://www.cefetmg.br/noticias/edital-seleciona-alunos-especiais-para-mestrado-em-automacao-e-sistemas/"
>
  <div
    className="News"
  >
    <ForwardRef(Divider)
      textAlign="left"
    >
      <MuiDividerRoot
        as="div"
        className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren MuiDivider-textAlignLeft"
        ownerState={
          Object {
            "absolute": false,
            "children": <ForwardRef(Typography)
              sx={
                Object {
                  "color": "#5579b9",
                  "padding": "20px",
                }
              }
              variant="h4"
            >
              Graduação
            </ForwardRef(Typography)>,
            "component": "div",
            "flexItem": false,
            "light": false,
            "orientation": "horizontal",
            "role": "separator",
            "textAlign": "left",
            "variant": "fullWidth",
          }
        }
        role="separator"
      >
        <Noop />
        <div
          className="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren MuiDivider-textAlignLeft css-cppmpw-MuiDivider-root"
          role="separator"
        >
          <MuiDividerWrapper
            className="MuiDivider-wrapper"
            ownerState={
              Object {
                "absolute": false,
                "children": <ForwardRef(Typography)
                  sx={
                    Object {
                      "color": "#5579b9",
                      "padding": "20px",
                    }
                  }
                  variant="h4"
                >
                  Graduação
                </ForwardRef(Typography)>,
                "component": "div",
                "flexItem": false,
                "light": false,
                "orientation": "horizontal",
                "role": "separator",
                "textAlign": "left",
                "variant": "fullWidth",
              }
            }
          >
            <Noop />
            <span
              className="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
            >
              <ForwardRef(Typography)
                sx={
                  Object {
                    "color": "#5579b9",
                    "padding": "20px",
                  }
                }
                variant="h4"
              >
                <MuiTypographyRoot
                  as="h4"
                  className="MuiTypography-root MuiTypography-h4"
                  ownerState={
                    Object {
                      "align": "inherit",
                      "children": "Graduação",
                      "className": undefined,
                      "color": undefined,
                      "component": undefined,
                      "gutterBottom": false,
                      "noWrap": false,
                      "paragraph": false,
                      "sx": Object {
                        "color": "#5579b9",
                        "padding": "20px",
                      },
                      "variant": "h4",
                      "variantMapping": Object {
                        "body1": "p",
                        "body2": "p",
                        "h1": "h1",
                        "h2": "h2",
                        "h3": "h3",
                        "h4": "h4",
                        "h5": "h5",
                        "h6": "h6",
                        "inherit": "p",
                        "subtitle1": "h6",
                        "subtitle2": "h6",
                      },
                    }
                  }
                  sx={
                    Object {
                      "color": "#5579b9",
                      "padding": "20px",
                    }
                  }
                >
                  <Noop />
                  <h4
                    className="MuiTypography-root MuiTypography-h4 css-h2oc99-MuiTypography-root"
                  >
                    Graduação
                  </h4>
                </MuiTypographyRoot>
              </ForwardRef(Typography)>
            </span>
          </MuiDividerWrapper>
        </div>
      </MuiDividerRoot>
    </ForwardRef(Divider)>
    <ForwardRef(Typography)
      sx={
        Object {
          "color": "#5579b9",
          "padding": "20px",
        }
      }
      variant="h2"
    >
      <MuiTypographyRoot
        as="h2"
        className="MuiTypography-root MuiTypography-h2"
        ownerState={
          Object {
            "align": "inherit",
            "children": "Edital seleciona alunos especiais para Mestrado em Automação e Sistemas",
            "className": undefined,
            "color": undefined,
            "component": undefined,
            "gutterBottom": false,
            "noWrap": false,
            "paragraph": false,
            "sx": Object {
              "color": "#5579b9",
              "padding": "20px",
            },
            "variant": "h2",
            "variantMapping": Object {
              "body1": "p",
              "body2": "p",
              "h1": "h1",
              "h2": "h2",
              "h3": "h3",
              "h4": "h4",
              "h5": "h5",
              "h6": "h6",
              "inherit": "p",
              "subtitle1": "h6",
              "subtitle2": "h6",
            },
          }
        }
        sx={
          Object {
            "color": "#5579b9",
            "padding": "20px",
          }
        }
      >
        <Noop />
        <h2
          className="MuiTypography-root MuiTypography-h2 css-gydxw7-MuiTypography-root"
        >
          Edital seleciona alunos especiais para Mestrado em Automação e Sistemas
        </h2>
      </MuiTypographyRoot>
    </ForwardRef(Typography)>
    
    
    <h3
      key="1"
    >
      O content vai ser pego pelo crawler usando a url
    </h3>
    
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Corrupti cum praesentium vero incidunt quod non quae, laudantium fuga eos dignissimos eum aliquam, aspernatur fugiat ipsa quos quisquam deleniti velit maiores et, eligendi consectetur numquam. Repellat fugit dolor hic veritatis voluptatibus consequuntur praesentium consectetur, labore, nihil quae itaque! Eius eum, quasi odio voluptatem voluptatum nam similique aut explicabo! Tenetur provident, reiciendis quis saepe aut modi voluptatum, blanditiis dignissimos distinctio accusantium dolores.
  
  </div>
</News>
`;
